app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -298,21 +314,38 @@ private void updateNavigationMenuSelection() {
         }
     }
 
-    private void addRoomColumns(HorizontalSnapScrollView scroller) {
-        LinearLayout root = (LinearLayout) scroller.getChildAt(0);
-        root.removeAllViews();
+    private void addRoomColumns(HorizontalSnapScrollView scroller, List<Lecture> lectures) {
         if (scroller.getColumnWidth() != 0) {
             // update pre-calculated width with actual layout
             columnWidth = scroller.getColumnWidth();
         }
-        for (int i = 0; i < MyApp.roomCount; i++) {
-            LinearLayout roomLayout = new LinearLayout(context);
-            LinearLayout.LayoutParams p = new LayoutParams(
-                    columnWidth, LayoutParams.MATCH_PARENT, 1);
-            roomLayout.setOrientation(LinearLayout.VERTICAL);
-            roomLayout.setLayoutParams(p);
-            root.addView(roomLayout);
+
+        LinearLayout columnsLayout = (LinearLayout) scroller.getChildAt(0);
+        columnsLayout.removeAllViews();
+        Map<Integer, List<Lecture>> lecturesPerRoom = groupLecturesPerRoom(lectures);
+        int boxHeight = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
+        LayoutCalculator layoutCalculator = new LayoutCalculator(Logging.Companion.get(), boxHeight);
+
+        for (int roomIndex = 0; roomIndex < MyApp.roomCount; roomIndex++) {
+            boolean roomHasLecture = lecturesPerRoom.containsKey(roomIndex);
+            if (!roomHasLecture) continue;
+            List<Lecture> roomLectures = lecturesPerRoom.get(roomIndex);
+
+            Map<Lecture, LayoutParams> lecturesLayoutParams = layoutCalculator.calculateLayoutParams(roomIndex, roomLectures, conference);
```

Comment:
:mag: Please suppress Lint NPE warning for `roomLectures` here.

Reply:
Cant suppress this warning I guess.

Reply:
Just wait for Lint to highlight the issue and then press `ALT` + `ENTER` to get a quick fix. Then select `Suppress for statement with comment` as shown in the screenshot.

![image](https://user-images.githubusercontent.com/144518/80647943-93915680-8a6f-11ea-8ee7-704c00dd0eff.png)

This should add `//noinspection ConstantConditions` above the line.

Reply:
Even better: When you add `override fun get(roomIndex: Int) = super.get(roomIndex) ?: emptyList()` to `LecturesPerRoom` then you get rid of the nullable list.

Reply:
cool